libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1014,13 +1014,10 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 		case FunctionType::Kind::ArrayPop:
 		{
 			_functionCall.expression().accept(*this);
-			solAssert(function.parameterTypes().empty(), "");
-
-			ArrayType const& arrayType = dynamic_cast<ArrayType const&>(
-				*dynamic_cast<MemberAccess const&>(_functionCall.expression()).expression().annotation().type
-			);
+//			solAssert(function.bound(), "");
```

Comment:
Not sure why it is not bound here.

Reply:
Why should it be bound?

Reply:
@christeth because this is not `bound` here, it is not the exact same `FunctionType` object I created before, right? I think I missed something here. Maybe you can give a hint?